api4/team.go

```diff
@@ -1080,10 +1088,16 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	defer c.LogAuditRec(auditRec)
 	auditRec.AddMeta("team_id", c.Params.TeamId)
 	auditRec.AddMeta("count", len(emailList))
+	auditRec.AddMeta("emails", emailList)
 
 	if graceful {
 		invitesWithError, err := c.App.InviteNewUsersToTeamGracefully(emailList, c.Params.TeamId, c.App.Session().UserId)
 		if err != nil {
+			errList := make([]string, 0, len(invitesWithError))
```

Comment:
I think this should be executed only if `err == nil`, because the graceful approach is to return a valid invitesWithError or an error, but not both.

Reply:
Ah, good point.  This is where the recent `App is an interface` change causes some issues.  The hover help on the API doesn't work anymore, and it's harder to find the actually implementation to review comment/code.

Reply:
Upon further review I found that `App.AddTeamMembers` and `InviteNewUsersToTeamGracefully` both return all the users passed in, not just the errors.  Successful adds have a nil error within the returned struct.  I now filter on the embedded error in both cases.

Reply:
https://github.com/mattermost/mattermost-server/pull/13967/commits/7cd3ae008aec55253760faf5d9091ba27602beee should resolve this.